### PR TITLE
Cancel any pending function calls when unmount

### PR DESCRIPTION
**Pull request by Yukimotochern**

### CODE REVIEW

**Comment by imbhargav5** [comment]
@Yukimotochern This looks awesome. Could I ask you to add a test case for this? That would be awesome! Thanks! 🙏🏽

File: src/hooks/useDebounce.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ function useDebounce<T extends (...args: any[]) => unknown>(
 
   useEffect(() => {
     debouncedCallbackRef.current = createDebouncedCallback(callback);
+    return debouncedCallbackRef.current.cancel();
```
Comment (imbhargav5) [inline src/hooks/useDebounce.ts:35]:
```suggestion
    return () => debouncedCallbackRef.current?.cancel();
```

**Comment by Yukimotochern** [comment]
@imbhargav5 This is the correct answer. Thanks.

**Comment by imbhargav5** [comment]
@Yukimotochern Sounds good! I will get this merge once I get someone to review this. cc @qqpann .

**Comment by qqpann** [comment]
@imbhargav5  The code looks good to me!

**Comment by imbhargav5** [comment]
Closing as I have moved your commits to #1302

**Pull request closed without merging**